CHANGELOG.md

```diff
@@ -1,5 +1,19 @@
 # HSD Release Notes & Changelog
 
+## unreleased
+
+### Node changes
+
+- `FullNode` has new RPC command `compacttree` which reduces the Urkel Tree disk
+usage by deleting historical data. It will keep up to the last 288 blocks worth
+of tree data on disk (7-8 tree intervals) exposing the node to a similar deep
+reorganization vulnerability as a chain-pruning node.
+
+- `FullNode` parses new configuration option `--compact-tree` which will compact
+the Urkel Tree only when the node first opens. This is the preferred method
+because it will run before the node connects to the network and the processing
+time will not affect peers.
```

Comment:
Smart. Prune mode is for the block data and this is the urkel tree, so they're orthogonal to each other, right?

Reply:
Right, but obviously closely related, chain and tree rely on each other especially during reorgs. By keeping the tree-compactor to < prune amount we can always be able to prune the tree no matter what configuration the chain is in (confirmed by tests) and vice versa.

If you are NOT pruning (full archival chain) but have compacted the tree, you could theoretically recover from a >288-block reorg but I did not implement that, and I think it will require some kind of rebuild from genesis (using all data on disk)

Reply:
I see. Is that 288 blocks configurable anywhere or is it hard-coded? Seems like it might come in handy to have the option, e.g. within tests.

Reply:
In bcoin and hsd it's hard coded. Bitcoin core is more flexible but 288 is the minimum according to spec.

Reply:
minimum for pruning? there's no urkel tree equivalent in btc is there?

Reply:
https://github.com/bitcoin/bips/blob/master/bip-0159.mediawiki